third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rspamd.task;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.util.ReactorUtils;
+import org.apache.james.util.streams.Iterators;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class GetMailboxMessagesService {
+    private final MailboxManager mailboxManager;
+    private final UsersRepository userRepository;
+
+    public GetMailboxMessagesService(MailboxManager mailboxManager, UsersRepository userRepository) {
+        this.mailboxManager = mailboxManager;
+        this.userRepository = userRepository;
+    }
+
+    public Flux<MessageResult> getMailboxMessagesOfAllUser(String mailbox) throws UsersRepositoryException {
+        return Iterators.toFlux(userRepository.list())
+            .flatMap(username -> getMailboxMessagesOfAUser(username, mailbox), ReactorUtils.DEFAULT_CONCURRENCY);
+    }
+
+    private Flux<MessageResult> getMailboxMessagesOfAUser(Username username, String mailbox) {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
+
+        return Mono.from(mailboxManager.getMailboxReactive(MailboxPath.forUser(username, mailbox), mailboxSession))
+            .flatMapMany(messageManager -> Flux.from(messageManager.getMessagesReactive(MessageRange.all(), FetchGroup.FULL_CONTENT, mailboxSession)));
```

Comment:
The use of FULL_CONTENT here will load all the messages content, all the time.

This means that this task will always load ALL the content of SPAM mailbox, which can be avoided with sampling/period parameters.

I suggest only loading metadata here and loading content upon reporting ham/spam.